### ISSUE TITLE
Can old package versions be removed from drat folder?

### ISSUE THREAD

**Issue by MikkoVihtakari** [eddelbuettel/drat#133]
Thanks for a great package! This is more of a question, not an issue. I was not sure where to ask it since the question feels too specific for SO. Instruct another place, if you do not want to answer questions here.

I have a drat repository for a CRAN package which I use as a data package because of the 5 Mb limit. There are old zip/tgz files in the folders for old R versions. Since this is a data package, the files are quite large (30 Mbish), which feels unnecessary space use of GitHub. Is it safe to remove these old zip/tgz files? How do I get drat to use only one zip/tgz file for all R versions?

> tree -s
14 directories, 34 files
(base) HI-11155:drat a22357$ tree -s
[        288]  .
├── [        171]  README.md
├── [        160]  bin
│   ├── [        128]  macosx
│   │   └── [        192]  contrib
│   │       ├── [        192]  4.0
│   │       │   ├── [        145]  PACKAGES
│   │       │   ├── [        155]  PACKAGES.gz
│   │       │   ├── [        284]  PACKAGES.rds
│   │       │   └── [   32164238]  ggOceanMapsData_1.0.1.tgz
│   │       ├── [        192]  4.1
│   │       │   ├── [        143]  PACKAGES
│   │       │   ├── [        153]  PACKAGES.gz
│   │       │   ├── [        280]  PACKAGES.rds
│   │       │   └── [   20155141]  ggOceanMapsData_1.2.tgz
│   │       └── [        192]  4.2
│   │           ├── [        143]  PACKAGES
│   │           ├── [        154]  PACKAGES.gz
│   │           ├── [        280]  PACKAGES.rds
│   │           └── [   23766939]  ggOceanMapsData_1.4.tgz
│   └── [        128]  windows
│       └── [        224]  contrib
│           ├── [        192]  4.0
│           │   ├── [        104]  PACKAGES
│           │   ├── [        122]  PACKAGES.gz
│           │   ├── [        344]  PACKAGES.rds
│           │   └── [   32164862]  ggOceanMapsData_1.0.0.zip
│           ├── [        192]  4.1
│           │   ├── [        102]  PACKAGES
│           │   ├── [        120]  PACKAGES.gz
│           │   ├── [        340]  PACKAGES.rds
│           │   └── [   20162939]  ggOceanMapsData_1.2.zip
│           ├── [        192]  4.2
│           │   ├── [        102]  PACKAGES
│           │   ├── [        120]  PACKAGES.gz
│           │   ├── [        340]  PACKAGES.rds
│           │   └── [   20182397]  ggOceanMapsData_1.2.zip
│           └── [        192]  4.3
│               ├── [        102]  PACKAGES
│               ├── [        120]  PACKAGES.gz
│               ├── [        340]  PACKAGES.rds
│               └── [   23669981]  ggOceanMapsData_1.4.zip
├── [         35]  index.html
└── [        128]  src
    └── [        224]  contrib
        ├── [        164]  PACKAGES
        ├── [        167]  PACKAGES.gz
        ├── [        312]  PACKAGES.rds
        └── [   23321926]  ggOceanMapsData_1.4.tar.gz

**Comment by eddelbuettel** [comment]
To clarify, you want to remove the versions for R 4.0 or R 4.1, say?  

I don't think we have a command that 'travels' across the tree like that.  We have one that remove _within_ a given version and operating syste, combination.  

So here I would just recommend a quick `git rm ...` of the respective files.  Good enough?

**Comment by MikkoVihtakari** [comment]
Yes, I thought it could be nice to link R 4.0 and R 4.1 (etc.) folders with the newest file, but I guess R repositories do not work like that.

If one had older version of R, say, 4.0 and there was no zip/tgz file in that folder, would `install.packages()` use the newer version or would it just fail? I.e. that one has to have a separate folder for each subversion for R?

**Comment by eddelbuettel** [comment]
R will always go by its _version_ first (and only).  So R 4.0 and 4.1 do not see the newer version you have here.

So in essence the above is already as compact as it gets _given your choice of supporting that many R versions_.  I would just prune as many old and deprecated R versions as you deem appropriate and show a little README.md snippet of using `download.file()` and local installation on older versions as your package like is 'unversioned' and suitable for all.

But as `drat` follows what R does and "merely" facilitates R's own repo creation, there is not too much more we can do here.

**Comment by MikkoVihtakari** [comment]
Ok, thanks for the information. Can I replace the file in older R version folder with the new one and just run `pruneRepo()` or `updateRepo()` to update the index? I don't think the new CRAN version works with those old versions of the data package.

**Comment by eddelbuettel** [comment]
Good point. I would just try it.  I don't think R will bark (when creating the index data) that the package was made by a newer R.

You can also try to work with a 'source' package then you'd only need one package per R version (and no windows/macOS builds) if you tell your users to add `type="source"` to `install.packages()`.  Default for me on Linux...

**Comment by MikkoVihtakari** [comment]
Thanks. I'll give it a try once the CRAN checks have dealt with a new update of the package.
